test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/ZookeeperService.java

```diff
@@ -0,0 +1,258 @@
+                                               .thenCompose(v -> k8Client.createAndUpdateCustomObject(CUSTOM_RESOURCE_GROUP, CUSTOM_RESOURCE_VERSION,
+                                                                                                      NAMESPACE, CUSTOM_RESOURCE_PLURAL,
+                                                                                                      getZookeeperDeployment(getID(), DEFAULT_INSTANCE_COUNT))),
+                                       t -> new TestFrameworkException(RequestFailed, "Failed to deploy zookeeper operator/service", t));
+        if (wait) {
+            Futures.getAndHandleExceptions(k8Client.waitUntilPodIsRunning(NAMESPACE, "app", ID, DEFAULT_INSTANCE_COUNT),
+                                           t -> new TestFrameworkException(RequestFailed, "Failed to deploy zookeeper service", t));
+        }
+    }
+
+
+    @Override
+    public void stop() {
+        Futures.getAndHandleExceptions(k8Client.deleteCustomObject(CUSTOM_RESOURCE_GROUP, CUSTOM_RESOURCE_VERSION, NAMESPACE, CUSTOM_RESOURCE_PLURAL, getID()),
+                                       t -> new TestFrameworkException(RequestFailed, "Failed to stop zookeeper service", t));
+    }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public String getID() {
+        return ID;
+    }
+
+    @Override
+    public boolean isRunning() {
+
+        return k8Client.getStatusOfPodWithLabel(NAMESPACE, "app", ID)
+                       .thenApply(statuses -> statuses.stream()
+                                                      .filter(podStatus -> podStatus.getContainerStatuses()
+                                                                                    .stream()
+                                                                                    .allMatch(st -> st.getState().getRunning() != null))
+                                                      .count())
+                       .thenApply(runCount -> runCount == DEFAULT_INSTANCE_COUNT)
+                       .exceptionally(t -> {
+                           log.warn("Exception observed while checking status of pod " + ID, t);
+                           return false;
+                       }).join();
+    }
+
+    @Override
+    public List<URI> getServiceDetails() {
+        //fetch the URI.
+        return Futures.getAndHandleExceptions(k8Client.getStatusOfPodWithLabel(NAMESPACE, "app", ID)
+                                                      .thenApply(statuses -> statuses.stream().map(s -> URI.create(TCP + s.getPodIP() + ":" + ZKPORT))
+                                                                                     .collect(Collectors.toList())),
+                                              t -> new TestFrameworkException(RequestFailed, "Failed to fetch ServiceDetails for Zookeeper", t));
+    }
+
+    @Override
+    public CompletableFuture<Void> scaleService(int instanceCount) {
```

Comment:
I haven't been able to make scaling work with the Pravega operator. There is an issue open for it.

Reply:
The zookeeper operator supports it and multiple zk pods are created by the operator(none of the system tests use scaleService for zookeeper).